heartbeat/beater/heartbeat.go

```diff
@@ -67,17 +62,14 @@ func New(b *beat.Beat, rawConfig *conf.C) (beat.Beater, error) {
 		return nil, fmt.Errorf("error reading config file: %w", err)
 	}
 
-	stateLoader := monitorstate.NilStateLoader
-
-	if b.Config.Output.Name() == "elasticsearch" {
-		// Connect to ES and setup the State loader
-		esc, err := getESClient(b.Config.Output.Config())
-		if err != nil {
-			return nil, err
-		}
-		if esc != nil {
-			stateLoader = monitorstate.MakeESLoader(esc, "synthetics-*,heartbeat-*", parsedConfig.RunFrom)
+	stateLoader, replaceStateLoader := monitorstate.AtomicStateLoader(monitorstate.NilStateLoader)
+	if b.Config.Output.Name() == "elasticsearch" && !b.Manager.Enabled() {
```

Comment:
We should also add in the changes from https://github.com/elastic/beats/pull/33375/files#diff-3edc3a1c86f948ac04dc2aeef09be1c549a2e5cbedd80a76e49dde4d95fa6883R82 which adds a `heartbeat.states.enabled` config flag. The service won't have the manager enabled, so we'll have to enable that to have states work with the service.

Reply:
This actually checks if heartbeat is in unmanaged mode (`!b.Manager.Enabled()`) to initialize a non-deferred states loader, so it should apply for the service as well.

IIRC, states flag was motivated by https://github.com/elastic/beats/issues/33357. IMO, this PR supersedes the original problem since we can now load the config on both modes. Is there an scenario where we would still like to disable states?